mk/aosp_optee.mk

```diff
@@ -115,6 +115,10 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_REQUIRED_MODULES := $(local_module_deps)
 
 TA_TMP_DIR := $(subst /,_,$(LOCAL_PATH))
+ifneq ($(filter-out $(OPTEE_TA_TARGETS),$(local_optee_ta_target)),)
+TA_DEV_KIT_DIR := $(TA_DEV_KIT_DIR:$(OPTEE_TA_TARGETS)=$(local_optee_ta_target))
+TA_TMP_DIR := $(addsuffix _$(local_optee_ta_target), $(TA_TMP_DIR))
```

Comment:
Why is this needed?

Reply:
For a common static library that could be linked by several TAs (in different architectures), we would put the intermediate object and library files in different architecture to different folder. Also, it helps the TA in specified architecture (different from the default one) to find the correct folder of the static library in the same architecture.

Reply:
OK

Reply:
Would that be acceptable if we also change the original path by adding the `"_ta_{target_arch}"` to the `$(TA_TMP_DIR)` no matter `$(local_optee_ta_target)` holds a value? like,

```
ifneq ($(local_optee_ta_target),)
TA_TMP_DIR := $(addsuffix _$(local_optee_ta_target), $(subst /,_,$(LOCAL_PATH)))
else
TA_TMP_DIR := $(addsuffix _$(OPTEE_TA_TARGETS), $(subst /,_,$(LOCAL_PATH)))
endif
```

Reply:
Sure. In fact that should allow using a helper variable instead of `local_optee_ta_target` and `OPTEE_TA_TARGETS`.

Reply:
Good idea. Updated.